04-rendering-output.Rmd

```diff
@@ -107,6 +107,19 @@ At a lower level, the arguments inside the parentheses will be used as the argum
 
 While this approach can be used to statically define the size of an image, it will not work for dynamic sizing of an image. If you wish to dynamically size images, you will need render and capture the graphical output yourself and return the contents with the appropriate `Content-Type` header. See the existing image renderers as a model of how to do this.
 
+### Return a file
+
+Sometimes it is good to have an endpoint where a particular file lives. This could be a [mounted](#mount-static) directory where serveral files would live and querying any of these would "save" the file at the front end. However, this could be an R object that is generated on the fly and returned at a particular `URL`. For example, `http:localhost:8000/api/docs/users.csv`. This could be achieved by: 
+
+```{r, eval=FALSE}
+#* @serializer contentType list(type="text/csv")
+#* @get /api/docs/users.csv # the endpoint.
+api.get_report <- function () {
+    [...]
+# return csv content
```

Comment:
Instead of leaving this an exercise, would it be possible to show an example of rendering a real CSV (or we could switch to plaintext if you think the CSV render is going to be too complicated).

But I imagine it would be a useful example to show how to serialize a base function (like write.csv) to an in-memory connection that could then be serialized to the client without having to touch the disk.

Reply:
Sounds good, will edit the commit.